### ISSUE TITLE
Category listing

### ISSUE THREAD

**Issue by c-martinez** [esciencecenter-digital-skills/NEBULA#11]
At the moment, the metadata of the modules includes the `category` keyword, which is used to define in which category the module will appear. But if the category does not exist, it will silently fail (for example if your category is `Getting started` instead of `Getting Started`).

As a user, it would be nice if "unknown categories" were added automatically to the list of existing categories (maybe thrown at the end).

**Comment by JaroCamphuijsen** [comment]
I would like to review the whole categorization and filtering of modules. 
An interesting example I find very useful is the github projects filtering, grouping and sorting:

In the example below, I filtered issues on being in the "previous iteration" (an iteration is a period of time you define that you work on a project, like a sprint or a coworking day), I grouped them by repository (where the issue belongs to) and sorted them according to their "priority" (which is just a numerical label between 1 and 4). 

![image](https://github.com/esciencecenter-digital-skills/NEBULA/assets/12695232/2841cc1c-e682-45d7-840d-8bf2ee97cd8d)

I think this flexibility of filtering, grouping and sorting is too complicated for us, but it is nice as an example.
In our case the filtering could be based on a label, which defines an interesting subset, like "masterclass", "life science", "technology" etc. Sorting could be according to a preset "order" much like we currently do for the chapters, and grouping might still be according to the current categories, or using some other grouping. Perhaps having a set of useful "collections" which have preset filters, groupings and sortings is what we actually need. 

These could perhaps be defined in the config.json file e.g.:
```json
"collections": {
  "masterclass": {
    "filter": "masterclass",
    "sort": "order",
    "group": "category"
  }
}
```

These collections could then be listed in a clear spot on the main page.
Or is this already too complicated and do we just need the proper labels to be able to filter and get rid of all the complexity?

**Comment by JaroCamphuijsen** [comment]
@raar1 @c-martinez @Cmurilochem what do you think?

**Comment by c-martinez** [comment]
I am not sure if the collection filtering would solve the initial problem: I add a new module, but mistype the category name. As a user I only see the new module is not showing at all, and have no idea where the problem is.

I am not against if it is a feature that would be useful in other circumstances. I would just ask what happens if I mistype the category name? Where does my module ends up? If it doesn't show up at all (because it doesn't match the filter), I think the user would still find it frustrating.

**Comment by JaroCamphuijsen** [comment]
yes @c-martinez sorry, I understand the original issue that you mentioned and I agree it should definitely be solved. I just wanted to mention that we might want to rethink and replace the whole category grouping system or at least extend it with the ability to let the user filter, group and sort modules themselves. But I actually think that it belongs in a different issue.
